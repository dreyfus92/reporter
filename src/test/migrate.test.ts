import {describe, it, expect, beforeAll, afterAll} from 'vitest';
import path from 'node:path';
import fs from 'node:fs/promises';
import {
  createTempDir,
  cleanupTempDir,
  runCliProcess,
  stripVersion
} from './utils.js';

let tempDir: string;

beforeAll(async () => {
  // Create a temporary directory for the test
  tempDir = await createTempDir();

  // Copy the basic-chalk fixture to the temp dir
  const fixturePath = path.join(process.cwd(), 'test/fixtures/basic-chalk');
  await fs.cp(fixturePath, tempDir, {recursive: true});
});

afterAll(async () => {
  await cleanupTempDir(tempDir);
});

describe('migrate command', () => {
  it('should handle --all flag correctly', async () => {
    const {stdout, stderr, code} = await runCliProcess(
      ['migrate', '--all', '--dry-run'],
      tempDir
    );

    if (code !== 0) {
      console.error('CLI Error:', stderr);
    }

    expect(code).toBe(0);
    expect(stripVersion(stdout)).toMatchSnapshot();
    expect(stderr).toBe('');
  });

  it('should handle specific package migration', async () => {
    const {stdout, stderr, code} = await runCliProcess(
      ['migrate', 'chalk', '--dry-run'],
      tempDir
    );

    expect(code).toBe(0);
    expect(stripVersion(stdout)).toMatchSnapshot();
    expect(stderr).toBe('');
  });

  it('should handle interactive mode', async () => {
    // Test interactive mode by providing input to the prompt
    // Press Enter to accept the default selection
    const {stdout, stderr, code} = await runCliProcess(
      ['migrate', '--interactive', 'chalk', '--dry-run'],
      tempDir,
      '\n' // Press Enter to accept default
    );

    expect(code).toBe(0);
    expect(stripVersion(stdout)).toMatchSnapshot();
    expect(stderr).toBe('');
  });

  it('should handle custom include pattern', async () => {
    const {stdout, stderr, code} = await runCliProcess(
      ['migrate', 'chalk', '--include', '**/*.js', '--dry-run'],
      tempDir
    );

    expect(code).toBe(0);
    expect(stripVersion(stdout)).toMatchSnapshot();
    expect(stderr).toBe('');
  });
});
