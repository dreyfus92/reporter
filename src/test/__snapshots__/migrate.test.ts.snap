// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`migrate command > should actually modify files when not using --dry-run 1`] = `
"[36me18e[39m (cli <version>)

[90mâ”Œ[39m  Migrating packages...
[90mâ”‚[39m
[90mâ”‚[39m  Targets: [2mchalk[22m
[90mâ”‚[39m
[32mâ—‡[39m  {cwd}/lib/main.js...
[90mâ”‚[39m
[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[90mâ”‚[39m  [2mwriting {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m
[32mâ—†[39m  {cwd}/lib/main.js [2m(1 migrated)[22m
[90mâ”‚[39m
[90mâ””[39m  Migration complete.

"
`;

exports[`migrate command > should actually modify files when not using --dry-run 2`] = `
"import * as pc from 'picocolors';

console.log(pc.cyan('I am _so_ cyan'));
"
`;

exports[`migrate command > should handle --all flag correctly 1`] = `
"[36me18e[39m (cli <version>)

[90mâ”Œ[39m  Migrating packages...
[90mâ”‚[39m
[90mâ”‚[39m  Targets: [2mchalk[22m
[90mâ”‚[39m
[32mâ—‡[39m  {cwd}/lib/main.js...
[90mâ”‚[39m
[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[90mâ”‚[39m  [2mwriting {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m
[32mâ—†[39m  {cwd}/lib/main.js [2m(1 migrated)[22m
[90mâ”‚[39m
[90mâ””[39m  Migration complete.

"
`;

exports[`migrate command > should handle --all flag correctly 2`] = `
"import chalk from 'chalk';

console.log(chalk.cyan('I am _so_ cyan'));
"
`;

exports[`migrate command > should handle custom include pattern 1`] = `
"[36me18e[39m (cli <version>)

[90mâ”Œ[39m  Migrating packages...
[90mâ”‚[39m
[90mâ”‚[39m  Targets: [2mchalk[22m
[90mâ”‚[39m
[32mâ—‡[39m  {cwd}/lib/main.js...
[90mâ”‚[39m
[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[90mâ”‚[39m  [2mwriting {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m
[32mâ—†[39m  {cwd}/lib/main.js [2m(1 migrated)[22m
[90mâ”‚[39m
[90mâ””[39m  Migration complete.

"
`;

exports[`migrate command > should handle custom include pattern 2`] = `
"import chalk from 'chalk';

console.log(chalk.cyan('I am _so_ cyan'));
"
`;

exports[`migrate command > should handle interactive mode 1`] = `
"[36me18e[39m (cli <version>)

[90mâ”Œ[39m  Migrating packages...
[?25l[90mâ”‚[39m
[36mâ—†[39m  Select packages to migrate

[36mâ”‚[39m  [2mSearch:[22m [7m[8m_[28m[27m
[36mâ”‚[39m  [32mâ—¼[39m chalk
[36mâ”‚[39m  [2m[2mâ†‘/â†“[22m[2m to navigate â€¢ [2mSpace:[22m[2m select â€¢ [2mEnter:[22m[2m confirm â€¢ [2mType:[22m[2m to search[22m
[36mâ””[39m"
`;

exports[`migrate command > should handle interactive mode 2`] = `
"import chalk from 'chalk';

console.log(chalk.cyan('I am _so_ cyan'));
"
`;

exports[`migrate command > should handle specific package migration 1`] = `
"[36me18e[39m (cli <version>)

[90mâ”Œ[39m  Migrating packages...
[90mâ”‚[39m
[90mâ”‚[39m  Targets: [2mchalk[22m
[90mâ”‚[39m
[32mâ—‡[39m  {cwd}/lib/main.js...
[90mâ”‚[39m
[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m  [2mloading {cwd}/lib/main.js[22m
[90mâ”‚[39m  [2mmigrating chalk to picocolors[22m
[90mâ”‚[39m  [2mwriting {cwd}/lib/main.js[22m
[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[90mâ”‚[39m
[32mâ—†[39m  {cwd}/lib/main.js [2m(1 migrated)[22m
[90mâ”‚[39m
[90mâ””[39m  Migration complete.

"
`;

exports[`migrate command > should handle specific package migration 2`] = `
"import chalk from 'chalk';

console.log(chalk.cyan('I am _so_ cyan'));
"
`;
